src/string_radix_sort.jl

```diff
@@ -0,0 +1,386 @@
+        skipbytes += 2
+        remaining_bytes_to_load -= 2
+        shift_for_padding -= 2*8
+    end
+    if  remaining_bytes_to_load >= 1
+        res |= Base.zext_int(T, ntoh(unsafe_load(Ptr{UInt8}(pointer(s, skipbytes+1))))) << (shift_for_padding - 1*8)
+        skipbytes += 1
+        remaining_bytes_to_load -= 1
+        shift_for_padding -= 1*8
+    end
+
+    return res
+end
+
+function load_bits_with_padding(::Type{UInt16}, s::String, skipbytes = 0)  
+    n = sizeof(s)
+    T = UInt16
+    ns = (sizeof(T) - min(sizeof(T), n - skipbytes))*8
+    remaining_bytes_to_load = sizeof(T) - ns÷8
+    # start
+    res = zero(T)
+    shift_for_padding = sizeof(T)*8
+        
+    if  remaining_bytes_to_load >= 1
+        res |= Base.zext_int(T, ntoh(unsafe_load(Ptr{UInt8}(pointer(s, skipbytes+1))))) << (shift_for_padding - 1*8)
+        skipbytes += 1
+        remaining_bytes_to_load -= 1
+        shift_for_padding -= 1*8
+    end
+
+    return res
+end
+
+function load_bits(::Type{T}, s::String, skipbytes = 0)::T where T <: Unsigned
+    n = sizeof(s)
+    if n < skipbytes
+        res = zero(T)
+    elseif n - skipbytes >= sizeof(T)
+        res = ntoh(unsafe_load(Ptr{T}(pointer(s, skipbytes+1))))
+    else
+        res = load_bits_with_padding(T, s, skipbytes)
+    end
+    return res
+end
+
+
+# Radix sort for strings
+function sort!(svec::AbstractVector{String}, ::StringRadixSortAlg, o::Perm)
+    sort!(svec, 1, length(svec), StringRadixSort, o)
+end
+
+function sort!(svec::AbstractVector, lo::Int, hi::Int, ::StringRadixSortAlg, o::O) where O <: Union{ForwardOrdering, ReverseOrdering, LexicographicOrdering, Perm}
+    if isa(o, Perm)
+        if eltype(o.data) != String
+            throw(ArgumentError("Cannot use StringRadixSort on type $(eltype(o.data))"))
+        end
+        return sortperm_radixsort(o.data, order = o.order)
+    else
+        sort!(svec, lo, hi, StringRadixSort, o)
+    end
+end
+
+function sort!(svec::AbstractVector{String}, lo::Int, hi::Int, ::StringRadixSortAlg, o::O) where O <: Union{ForwardOrdering, ReverseOrdering, LexicographicOrdering}
+    #  Input checking
+    # this should never be the case where `o isa Perm` but the `svec`` is a string
+    # if isa(o, Perm)
+    #     throw(ArgumentError("Cannot use StringRadixSort on type $(eltype(o.data))"))
+        # if eltype(o.data) != String
+        #     throw(ArgumentError("Cannot use StringRadixSort on type $(eltype(o.data))"))
+        # end
+        # o = o.order
+        # svec = o.data
+        # return
+    # end
+    
+    if lo >= hi;  return svec;  end
+
+    # find the maximum string length    
+    lens = maximum(sizeof, svec)
+    skipbytes = lens
+    while lens > 0
+       if lens > 8
+            skipbytes = max(0, skipbytes - 16)
+            if o == Reverse
+                sorttwo!(.~load_bits.(UInt128, svec, skipbytes), svec)
+            else
+                sorttwo!(load_bits.(UInt128, svec, skipbytes), svec)
```

Comment:
Note that this allocates arrays over and over; would be better to allocate a `bits` array once and overwrite it in-place with `bits .= load_bits.(...)`